### ISSUE TITLE
lack of space in  config/action.d/firewallcmd-ipset.conf breaks creation of ipsets

### ISSUE THREAD

**Issue by cybernard** [fail2ban/fail2ban#2579]
config/action.d/firewallcmd-ipset.conf
```
actionstart = ipset -! create <ipmset> hash:ip timeout <bantime><familyopt> 
```
Notice a lack of space symbol between the \> and \< this causes the creation of ALL ipset to fail.
```diff
-ipset create f2b-postfix-sasl hash:ip timeout  600family inet
+ipset create f2b-postfix-sasl hash:ip timeout  600 family inet 
```

**Comment by sebres** [comment]
Default `familyopt` is either empty (ipv4) or contains space `<sp>` at begin (ipv6).
See https://github.com/fail2ban/fail2ban/blob/309a1cb337604e03f764bf50839bdd3cb8280757/config/action.d/firewallcmd-ipset.conf#L55
But it's indeed not so good if one uses own `familyopt` tag without space tag (do you?)

**Comment by cybernard** [comment]
The real issue here is without the space you have "timeout 600family"  this command is invalid because the 600 and family as mashed together without a space.

**Comment by sebres** [comment]
I'm not against to change this, but prio of the task is lowered now (I've more important things to do).

As I already said, it looks like your own config issue.
You write that you got this (for IPv4):
```
ipset create f2b-postfix-sasl hash:ip timeout 600family inet
```
But as you can see in https://github.com/fail2ban/fail2ban/blob/309a1cb337604e03f764bf50839bdd3cb8280757/config/action.d/firewallcmd-ipset.conf#L50
`familyopt` is empty for ipv4 family. So I'll expect:
```diff
- ipset create f2b-postfix-sasl hash:ip timeout 600family inet
+ ipset create f2b-postfix-sasl hash:ip timeout 600
```
In your case it looks like somewhere it'd set to `familyopt = family inet`. 
Change it to `familyopt = <sp>family inet` and it would work.

**Comment by cyb3rz3us** [comment]
Just curious - why wouldn't the best choice be to actually change the 'firewallcmd-ipset.conf' file?  Is there ever a valid use case where there is no space between '< bantime >' and '< familyopt >'?

**Comment by sebres** [comment]
> Is there ever a valid use case where there is no space...

I don't know the reason anymore, may be it is just an artificial thing (to avoid some large change of covering tests) or even something else like rebase from other branch where `<sp>` was expected.

> It seems odd to require a leading space for a config option.

Still again - I'm not against at all. And I'll change that. Do you see the issue is open yet?

**Comment by cyb3rz3us** [comment]
> I don't know the reason anymore

Hmmm...fair enough...maybe just a legacy situation then.  In any event, looks like it's gonna get modified to have the space in the 'actionstart' statement which should help avoid any further confusions...

**Comment by sebres** [comment]
Will be fixed in #2703 (as I anyway touched there every ipset-action as well as related test cases).
